lib/rules/require-each-key.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const AstNodeInfo = require('../helpers/ast-node-info');
+const Rule = require('./base');
+
+const ERROR_MESSAGE = '{{#each}} block requires a valid key value to avoid performance issues';
+
+const SPECIAL_KEY_VALUES = new Set(['@index', '@identity']);
+
+module.exports = class RequireEachKey extends Rule {
+  visitor() {
+    return {
+      BlockStatement(node) {
+        let isEach = AstNodeInfo.isEach(node);
+        let keyPair = node.hash.pairs.find((p) => p.key === 'key');
+        let keyValue = keyPair && keyPair.value && keyPair.value.value;
+        let isSpecialKey = keyValue && keyValue.startsWith('@');
+        let isValidKey = isSpecialKey ? SPECIAL_KEY_VALUES.has(keyValue) : true;
```

Comment:
I don't think it's necessary to check for `@`, just `SPECIAL_KEY_VALUES.has(keyValue)` should be enough.